---
title: Reconciling single-cell multiomics modalitites using metacells 
author:
  - name: Léonard Hérault
    affil: 1,2
    email: leonard.herault@unil.ch
    orcid: 0000-0001-6499-2991
    main: true
  - name: Aurélie Gabriel
    affil: 1,2
  
affiliation:
  - num: 1
    address: Department of Oncology, Ludwig Institute for Cancer research, University of Lausanne
  - num: 2
    address: Swiss Institute of Bioinformatics
main_findings:
  - "**Metacell** unlock the full potential of single-cell **multiomics** data by increasing modality **correlation**"
  - '![](https://raw.githubusercontent.com/brentthorne/posterdown/master/images/betterhexlogo.png){.main_pic}'
logoleft_name: '![](https://raw.githubusercontent.com/brentthorne/posterdown/master/images/qr-code-black.png){.main-img-left}'
logoright_name: '![](https://raw.githubusercontent.com/brentthorne/posterdown/master/images/betterhexlogo.png){.main-img-right}'
output: 
  posterdown::posterdown_betterport:
    self_contained: false
    pandoc_args: --mathjax
    highlight: haddock
    number_sections: false
link-citations: true
bibliography: packages.bib
---

```{r, include=FALSE}
knitr::opts_chunk$set(results = 'asis',
                      echo = FALSE,
                      warning = FALSE,
                      tidy = FALSE,
                      message = FALSE,
                      fig.align = 'center',
                      out.width = "100%")
options(knitr.table.format = "html") 
```

```{r myplot, include=FALSE}
svg('myplot.svg')
plot(iris$Sepal.Length, iris$Sepal.Width)
dev.off()
```

# Introduction

The increased throughput of single-cell omics technologies enables researchers to study cell type-specific gene regulation at an unprecedented resolution. For cancer, these advances promise a better understanding of the tumor micro-environment (TME), with implications in precision medicine and cancer immunotherapy. These promises depend on the development of computational methods to cope with both the high sparsity of these data and the constant increase in cell and sample numbers 

To address these needs, the concept of metacells, defined as disjoint and homogeneous groups of cells with high similarity, was proposed to decrease the size and the sparsity of scRNA-seq data. Recently, our group has developed SuperCell [bilous_metacells_2022-1], a network based coarse graining framework in which highly similar cells are merged into metacells that can be used for quantitative downstream analyses, like clustering or differential expression. We are now extending this concept to other single-cell omics technologies and have also developed a new version of our algorithm to identify metacells in single-cell multiomics data that combine the measurement of different types of molecules (modalities) in the same single-cell.  

```{r, include=FALSE}
knitr::write_bib(c('posterdown', 'rmarkdown','pagedown'), 'packages.bib')
```

## Objectives

1. Adapt SuperCell algorithm to identify metacells on single-cells multiomics data
  * CITE-seq data : single cell measurement of RNA & Protein level 
  * 10X multiome : single cell Assay for Transposase-Accessible Chromatin (ATAC) and RNA measurement  
2. Analyze inter-modality feature correlations at the metacell level

# Methods

To identify metacell on multiomics data we take advantage of the Weighted Neighrest Neighbor Analysis of single-cells multiomics data recently introduced [2]. First, we compute separately for each modality an adapted low dimension embedding. From these embeddings we construct a weighted nearest neighbor graph on which we identify metacell with the walktrap algorithm. Finally single cell raw data from each modality are aggregated per metacells.

$$
$$
We tested first our workflow on CITE-seq data of bone marrow cells presenting transcriptome profiling and 25 protein abundance measurement for 30'000 cells. We also analyzed a 10X multiome (RNA + ATAC) dataset of 12'000 Peripheral Blood Monocuclear Cells (PBMCs). For each dataset we identified metacell at different gamma and compute inter-modality feature correlations:
  * gene expression and protein abundances for CITE-seq data
  * gene body accessibility and gene expression for 10X multiome data

**_Now on to the results!_**

<br>

# Results

Here you may have some figures to show off, bellow I have made a scatterplot with the infamous Iris dataset and I can even reference to the figure automatically like this, `Figure \@ref(fig:irisfigure)`, Figure \@ref(fig:irisfigure).

```{r, irisfigure, fig.cap='Here is a caption for the figure. This can be added by using the "fig.cap" option in the r code chunk options, see this [link](https://yihui.name/knitr/options/#plots) from the legend himself, [Yihui Xie](https://twitter.com/xieyihui).', out.width="80%"}
par(mar=c(2,2,0,1))
plot(x = iris$Sepal.Length, y = iris$Sepal.Width,
     col = iris$Species, pch = 19, xlab = "Sepal Length",
     ylab = "Sepal Width")
```

Maybe you want to show off some of that fancy code you spent so much time on to make that figure, well you can do that too! Just use the `echo=TRUE` option in the r code chunk options, Figure \@ref(fig:myprettycode)!

```{r myprettycode, echo=FALSE,collapse=TRUE, fig.cap='Boxplots, so hot right now!', fig.height=3.5, out.width="80%"}
#trim whitespace
par(mar=c(2,2,0,0))
#plot boxplots
boxplot(iris$Sepal.Width~iris$Species,
        col = "#008080", 
        border = "#0b4545",
        ylab = "Sepal Width (cm)",
        xlab = "Species")
```

How about a neat table of data? See, Table \@ref(tab:iristable):

```{r, iristable}
knitr::kable(
  iris[1:8,1:5], format = "html",
  caption = "A table made with the **knitr::kable** function.",
  align = "c", col.names = c("Sepal <br> Length",
                             "Sepal <br> Width",
                             "Petal <br> Length",
                             "Petal <br> Width",
                             "Species"),
  escape = FALSE)
```

# References
