<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1">



<title>SuperCellMultiomics_SCCL_2022.knit</title>







<!--
Font-awesome icons ie github or twitter
-->
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/all.css" integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf" crossorigin="anonymous">
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/brands.css" integrity="sha384-n9+6/aSqa9lBidZMRCQHTHKJscPq6NW4pCQBiMmHdUCvPN8ZOg2zJJTkC7WIezWv" crossorigin="anonymous">

<!--
Google fonts api stuff
-->
<link href='https://fonts.googleapis.com/css?family=Special Elite' rel='stylesheet'>
<link href='https://fonts.googleapis.com/css?family=Rasa' rel='stylesheet'>

<!--
Here are the required style attributes for css to make this poster work :)
-->
<style>
@page {
size: 36in 48in;
margin: 0;
padding: 0;
}
body {
margin: 0px;
padding: 0px;
width: 36in;
height: 48in;
text-align: justify;
font-size: 45px;
line-height: 1.05;
}
/* RMarkdown Class Styles */
/* center align leaflet map,
from https://stackoverflow.com/questions/52112119/center-leaflet-in-a-rmarkdown-document */
.html-widget {
margin: auto;
position: sticky;
margin-top: 2cm;
margin-bottom: 2cm;
}
.leaflet.html-widget.html-widget-static-bound.leaflet-container.leaflet-touch.leaflet-fade-anim.leaflet-grab.leaflet-touch-drag.leaflet-touch-zoom {
position: sticky;
width: 100%;
}
pre.sourceCode.r {
background-color: #dddddd40;
border-radius: 4mm;
padding: 4mm;
width: 75%;
margin: auto;
margin-top: 1em;
margin-bottom: 1em;
/* align-items: center; */
}
code.sourceCode.r{
background-color: transparent;
font-size: 20pt;
border-radius: 2mm;
}
code {
font-size: 25pt;
font-family: monospace;
background-color: #00808024;
color: #0b4545;
padding: 1.2mm;
line-height: 1;
border-radius: 2mm;
}
caption {
margin-bottom: 10px;
font-size: 20pt;
font-style: italic;
}

tbody tr:nth-child(odd) {
    background-color: #00808020;
}
.table>thead>tr>th, .table>tbody>tr>th, .table>tfoot>tr>th, .table>thead>tr>td, .table>tbody>tr>td, .table>tfoot>tr>td{
  border-spacing: 0;
  font-size: 40%;
  border-style: none;
  padding-top: 15px;
  padding-bottom: 15px;
  padding-right: 1em;
  padding-left: 1em;
  line-height: 1em;
}
table {
  margin: auto;
}
th {
  padding-left: 5mm;
  padding-right: 5mm;
}
.caption {
font-size: 20pt;
font-style: italic;
padding-top: 0;
}
.references {
font-size: 20px;
line-height: 90%;
}
/* Create three unequal columns that floats next to each other */
.column {
float: left;
padding: 0px;
}
.outer {
width: 36in;
height: calc(48in *  (1 - 0.12 - 9.5e-2 - 0.01) );
-webkit-column-count: 3; /* Chrome, Safari, Opera */
-moz-column-count: 3; /* Firefox */
column-count: 3;
-webkit-column-fill: auto;
-moz-column-fill: auto;
column-fill: auto;
column-gap: 0;
padding-left: 0cm;
padding-right: 0cm;
/* -webkit-column-rule-width: 50%;
-moz-column-rule-width: 50%;
column-rule-width: 50%; */
-webkit-column-rule-style: none;
-moz-column-rule-style: none;
column-rule-style: none;
-webkit-column-rule-color: black;
-moz-column-rule-color: black;
column-rule-color: black;
background-color: #ffffff;
font-family: Rasa;
margin-top: calc(48in *  0.12 );
padding-top: 1em;
padding-bottom: 1em;
}
span.citation {
  color: #008080;
  font-weight: bold;
}
a {
text-decoration: none;
color: #008080;
}
#title {
font-size: 125pt;
text-align: left;
margin: 0;
line-height: 98%;
border-bottom: 0;
font-weight: normal;
background: 0;
}
#author {
color: #0b4545;
margin: 0;
line-height: 85%;
font-size: 1.17em;
}
#affiliation {
padding-top: 0.1em;
color: ;
font-style: italic;
font-size: 25px;
margin: 0;
}
sup {
color: #cc0000;
}
.affiliation sup {
font-size: 20px;
}
.author {
text-align: left;
}
.author sup {
font-size: 30px;
}
.author_extra {
color: #008080;
margin: 0;
line-height: 85%;
font-size: 35px;
text-align: left;
}
.outer h1, h2, h3, h4, h5, h6 {
text-align: center;
margin: 0;
font-weight: bold;
}
.section h1 {
  text-align:center;
  padding-bottom:5px;
  background:
    linear-gradient(
      to left,
      #ffffff 1%,
      #ffffff 20%,
      #0b454575 33%,
      #0b4545 50%,
      #0b454575 66%,
      #ffffff 80%,
      #ffffff 99%
    )
    left
    bottom
    #ffffff
    no-repeat;
  background-size:100% 5px ;
  margin-top: 0.5em;
  margin-bottom: 0.5em;
}
.outer h2 {
text-align: center;
}
.outer p, .level2 {
color: #000000;
}
.outer ol {
padding-left: 8%;
padding-right: 8%;
text-align: left;
}
.main {
width: 36in;
height: calc(48in * 0.12);
position: absolute;
background-color: #0b4545;
color: #ffffff90;
font-family: Special Elite;
background-image: linear-gradient(#0b4545 50%,#008080);
}
.main strong {
color: #ffffff;
}
.main strong > sup {
color: #ffffff;
}
.main sup {
color: #ffffff90;
}
#main-img-left {
width: 10%;
left: 0.5in;
bottom: 0.2in;
position: absolute;
}
#main-img-center {
width: 10%;
left: calc(36in * 0.45);
bottom: 0.5in;
position: absolute;
}
#main-img-right {
width: 10%;
right: 0.5in;
bottom: 0.2in;
position: absolute;
}
.main p {
font-size: 150px;
font-family: Special Elite;
text-align: center;
margin: 0;
position: absolute;
top: 50%;
-ms-transform: translateY(-50%);
transform: translateY(-50%);
margin-left: 1em;
}
.fab {
color: #00000030;
font-size: 25px;
}
.twitter, i {
color: #00000030;
font-size: 35px;
text-decoration: none;
}
a.email {
text-decoration: none;
color: #00000030;
font-size: 35px;
}
.envelope {
color: #00000030;
font-size: 5px;
text-decoration: none;
}
.poster_wrap {
width: 36in;
height: 48in;
padding: 0cm;
}
.main_bottom {
width: 36in;
height: calc(48in * 9.5e-2);
margin-top: calc(48in * (1 - 9.5e-2));
position: absolute;
background-color: #0b4545;
background-image: linear-gradient(#008080 10%, #0b4545);
}
.section {
  padding-left: 10mm;
  padding-right: 10mm;
}
span > #tab:mytable {
  font-weight: bold;
}
.orcid img {
  width: 3%;
}
.emphasis {
  background-color: #008080;
  color: #ffffff;
  border: solid #0b2045 3mm;
  margin: 1em;
  padding-left: 0;
  padding-right: 0;
}
.emphasis h1 {
  font-weight: bold;
  background: none;
  background-color: #0b2045;
  padding-bottom: 5mm;
  padding-top: 1mm;
  margin-top: -1mm;
  margin-right: -1mm;
  margin-left: -1mm;
}
.emphasis blockquote {
  border: 0;
}
.emphasis ol {
  padding: 0;
  padding-left: 8%;
  font-size: 100%;
  font-weight: bold;
}
.emphasis p {
  color: #ffffff;
}
</style>
</head>
<body>


<div class="poster_wrap">

<div class="column outer">
<div class="section">

<h3 id="author" class="author">

<strong>Léonard Hérault</strong><sup> 1,2, <a class="orcid" href="https://orcid.org/0000-0001-6499-2991"><img src="https://raw.githubusercontent.com/brentthorne/posterdown/master/images/orcid.jpg"></a></sup><br>

<a class='envelope'><i class="fas fa-envelope"></i></a> <a href="mailto:leonard.herault@unil.ch" class="email">leonard.herault@unil.ch</a> <br>
   </h3>

<h5 id="author_extra", class="author_extra">
 Aurélie Gabriel<sup>1,2</sup>
 Mariia Bilous<sup>1,2</sup>
 David Gfeller<sup>1,2</sup>
</h5>


<p id="affiliation" class="affiliation">
<sup>1</sup> Department of Oncology, Ludwig Institute for Cancer research, University of Lausanne<br> <sup>2</sup> Swiss Institute of Bioinformatics
</p>
</div>

<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>The increased throughput of single-cell omics technologies enables researchers to study cell type-specific gene regulation at an unprecedented resolution. These promises depend on the development of computational methods to cope with both large size and the high sparsity of these data. Thus, we and others developed methods to identify metacells, disjoint and homogeneous groups of cells, in scRNA-seq data <span class="citation">Bilous et al. (<a href="#ref-bilous_metacells_2022-1" role="doc-biblioref">2022</a>)</span>. We are now developing a new version of our method <strong>SuperCell</strong> to identify metacells in single-cell multiomics (sc-multiomics) data that combine the measurement of different types of molecules (modalities) in the same single-cell.</p>
<p><img src="images/FigureIntro.png.jpg" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="methods" class="section level1">
<h1>Methods</h1>
<p>For sc-multiomics, Supercell used a weighted Nearest Neighbor (wNN) graph <span class="citation">(<a href="#ref-hao_integrated_2021" role="doc-biblioref">Hao et al. 2021</a>)</span>), on which the metacell are identified. Single-cell raw data from each modality are aggregated per metacells that can be used for downstream analyses.
<img src="images/FigureMethod.jpg" width="100%" style="display: block; margin: auto;" /></p>
<div id="datasets" class="section level2">
<h2>Datasets</h2>
<ul>
<li>10X multiome data of Peripheral Blood Mononuclear cells (PBMCs) : scRNA-seq &amp; ATAC-seq (12’000 cells). We analysed the correlation between gene expression and gene activity.</li>
<li>CITE-seq data of Bone marrow cells: scRNA-seq &amp; 25 protein levels (30’000 cells). We analysed the correlation between gene expression and corresponding protein level.</li>
</ul>
<p><img src="images/FigureActivity.jpg" width="70%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="results" class="section level1">
<h1>Results</h1>
<div id="x-multiome-dataset-of-pbmc" class="section level3">
<h3>10X multiome dataset of pbmc</h3>
SuperCell identified robuste metacell in multiomic space of PBMC cells (Fig.<a href="#fig:pbmcmulti">1</a>).
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:pbmcmulti"></span>
<img src="SuperCellMultiomics_SCCL_2022_files/figure-html/pbmcmulti-1.png" alt="identified metacell in the multimomic space of PBMCs" width="100%" />
<p class="caption">
Figure 1: identified metacell in the multimomic space of PBMCs
</p>
</div>
<p><br />
</p>
<p>Gene activity and expression for key TF of lymphoid cells (TCF7) and myeloid cells (SPI1) appeared more correlated at the metacell level (Fig.<a href="#fig:crMultiomeKey">2</a>).</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:crMultiomeKey"></span>
<img src="SuperCellMultiomics_SCCL_2022_files/figure-html/crMultiomeKey-1.png" alt="gene expression-activity correlation for key TF of immune cells" width="100%" />
<p class="caption">
Figure 2: gene expression-activity correlation for key TF of immune cells
</p>
</div>
<p><br />
</p>
<p>Analysis of this correlation for the most variable genes at the transcriptomic level (top 2000) show the same trend (Fig.<a href="#fig:globalCrMultiome">3</a>).</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:globalCrMultiome"></span>
<img src="SuperCellMultiomics_SCCL_2022_files/figure-html/globalCrMultiome-1.png" alt="Global gene expression-activity correlation" width="100%" />
<p class="caption">
Figure 3: Global gene expression-activity correlation
</p>
</div>
<p><br />
<br />
<br />
<br />
</p>
</div>
<div id="cite-seq-dataset-of-bm-cells" class="section level3">
<h3>CITE-seq dataset of BM cells</h3>
<p>SuperCell identified robuste metacell in CITE-seq space of BM cells (Fig.<a href="#fig:bmcite">4</a>).</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:bmcite"></span>
<img src="SuperCellMultiomics_SCCL_2022_files/figure-html/bmcite-1.png" alt="identyfied metacell in the multimodal single cell CITE-seq space of bone marrow cells" width="100%" />
<p class="caption">
Figure 4: identyfied metacell in the multimodal single cell CITE-seq space of bone marrow cells
</p>
</div>
<p><br />
Metacell analysis increase the correlation between RNA and protein level of key markers of HSC (CD34) and CD8 T cells (CD8A) Fig.<a href="#fig:crBmKey">5</a>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:crBmKey"></span>
<img src="SuperCellMultiomics_SCCL_2022_files/figure-html/crBmKey-1.png" alt="RNA protein correlations for key markers of BM cells" width="90%" />
<p class="caption">
Figure 5: RNA protein correlations for key markers of BM cells
</p>
</div>
<p><br />
The analysis of the RNA-protein correlation for the all 25 gene-protein couples exhibits the same trend Fig.<a href="#fig:globalCrCiteSeq">6</a>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:globalCrCiteSeq"></span>
<img src="SuperCellMultiomics_SCCL_2022_files/figure-html/globalCrCiteSeq-1.png" alt="Global RNA-Protein correlations" width="100%" />
<p class="caption">
Figure 6: Global RNA-Protein correlations
</p>
</div>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-baran_metacell_2019" class="csl-entry">
Baran, Yael, Akhiad Bercovich, Arnau Sebe-Pedros, Yaniv Lubling, Amir Giladi, Elad Chomsky, Zohar Meir, Michael Hoichman, Aviezer Lifshitz, and Amos Tanay. 2019. <span>“<span>MetaCell</span>: Analysis of Single-Cell <span>RNA</span>-Seq Data Using <span>K</span>-Nn Graph Partitions.”</span> <em>Genome Biology</em> 20 (1): 206. <a href="https://doi.org/10.1186/s13059-019-1812-2">https://doi.org/10.1186/s13059-019-1812-2</a>.
</div>
<div id="ref-bilous_metacells_2022-1" class="csl-entry">
Bilous, Mariia, Loc Tran, Chiara Cianciaruso, Aurélie Gabriel, Hugo Michel, Santiago J. Carmona, Mikael J. Pittet, and David Gfeller. 2022. <span>“Metacells Untangle Large and Complex Single-Cell Transcriptome Networks.”</span> <em>BMC Bioinformatics</em> 23 (1): 336. <a href="https://doi.org/10.1186/s12859-022-04861-1">https://doi.org/10.1186/s12859-022-04861-1</a>.
</div>
<div id="ref-hao_integrated_2021" class="csl-entry">
Hao, Yuhan, Stephanie Hao, Erica Andersen-Nissen, William M. Mauck, Shiwei Zheng, Andrew Butler, Maddie J. Lee, et al. 2021. <span>“Integrated Analysis of Multimodal Single-Cell Data.”</span> <em>Cell</em> 184 (13): 3573–3587.e29. <a href="https://doi.org/10.1016/j.cell.2021.04.048">https://doi.org/10.1016/j.cell.2021.04.048</a>.
</div>
</div>
</div>

</div>
<div class="main">
<p>Reconciling <strong>single-cell multiomics</strong> modalities using <strong>metacells</strong></p>
</div>
<div class="main_bottom">
<img id="main-img-left" src=https://raw.githubusercontent.com/brentthorne/posterdown/master/images/betterhexlogo.png>
<img id="main-img-center" src=https://raw.githubusercontent.com/brentthorne/posterdown/master/images/qr-code-black.png>
<img id="main-img-right" src=https://raw.githubusercontent.com/brentthorne/posterdown/master/images/betterhexlogo.png>
</div>
</div>



</body>
</html>
